packages/api-service/src/api-service.ts

```diff
@@ -320,45 +191,114 @@ export default class ApiService {
       headers['Authorization'] = `Bearer ${token}`;
     }
 
-    let response;
+    let response: Response;
     try {
       response = await fetch(`${this.baseUrl}${urlPath}`, {
         method: 'GET',
         headers: headers,
         signal: options?.abortController?.signal,
       });
     } catch (error) {
-      if (error.message != null && error.message.indexOf('Failed to fetch') !== -1) {
-        return Promise.reject('Network error. Check your connection and try again.');
-      }
+      return Promise.reject(this.rewriteFetchErrors(error, 'GET', urlPath));
+    }
+
+    return await this.parseResponse(response, 'GET', urlPath);
+  }
+
+  private distinct<T>(value: T, index: number, array: T[]) {
+    return array.indexOf(value) === index;
+  }
+
+  public async aggregateResponses<T>(apiCalls: Promise<ODataResponse<T>>[]): Promise<void> {
+    const errorMessageToCount: Map<string, number> = new Map();
+    const httpErrors: HttpError[] = [];
 
-      if (error.name != null && error.name === 'AbortError') {
-        return Promise.reject('Abort error. Request has been aborted.');
+    const requests = apiCalls.map(async call => {
+      try {
+        await call;
+      } catch (httpError) {
+        httpErrors.push(httpError);
+        const errorMsg = httpError.message;
+        errorMessageToCount.set(errorMsg, (errorMessageToCount.get(errorMsg) ?? 0) + 1);
       }
+    });
 
-      return Promise.reject(error);
-    }
+    await Promise.allSettled(requests);
+
+    if (errorMessageToCount.size > 0) {
+      const newError: HttpError = {
+        type: 'HttpError',
+        action: httpErrors
+          .map(o => o.action)
+          .filter(this.distinct)
+          .join(', '),
+        message: `${httpErrors.length} of ${apiCalls.length} actions failed`,
+        detail: Array.from(errorMessageToCount.keys())
+          .map(o => `${errorMessageToCount.get(o)} error(s):  ${o}`)
+          .join('\n'),
+        baseUrl: httpErrors
+          .map(o => o.baseUrl)
+          .filter(this.distinct)
+          .join(', '),
+        path: httpErrors
+          .map(o => o.path)
+          .filter(this.distinct)
+          .join(', '),
+      };
+      return Promise.reject(newError);
+    }
+    return;
+  }
 
-    if (response.status === 404) {
-      return Promise.reject('404: Endpoint not found.');
-    }
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  private rewriteFetchErrors(error: any, action: string, urlPath: string, statusCode: number | undefined = undefined) {
+    const message = error?.message?.includes('Failed to fetch')
+      ? 'Network error. Check your connection and try again.'
+      : error?.name === 'AbortError'
+      ? 'Abort error. Request has been aborted.'
+      : error;
```

Comment:
should this be `: error?.message ?? error` in case error is an object here?

Reply:
👍🏻